src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -1889,22 +1889,20 @@ int CpuemShim::xclCloseContext(const uuid_t xclbinId, unsigned int ipIndex) cons
 
 void CpuemShim::constructQueryTable() {
   if (xclemulation::config::getInstance()->getIsPlatformEnabled()) {
-    mQueryTable["m2m"] = mPlatformData.get<std::string>("plp.m2m");
+    mQueryTable[key_type::m2m] = mPlatformData.get<std::string>("plp.m2m");
     std::string dmaVal = mPlatformData.get<std::string>("plp.dma");
-    mQueryTable["nodma"] = (dmaVal == "none" ? "enabled" : "disabled");
+    mQueryTable[key_type::nodma] = (dmaVal == "none" ? "enabled" : "disabled");
   }
 }
 
-bool CpuemShim::isM2MEnabled() {
+bool CpuemShim::deviceQuery(key_type queryKey) {
   if (xclemulation::config::getInstance()->getIsPlatformEnabled()) {
-    return (mQueryTable["m2m"] == "enabled" ? true : false);
-  }
-  return false;
-}
-
-bool CpuemShim::isNoDMAEnabled() {
-  if (xclemulation::config::getInstance()->getIsPlatformEnabled()) {
-    return (mQueryTable["nodma"] == "enabled" ? true : false);
+    if (queryKey == key_type::nodma) {
```

Comment:
While you are here, can you remove the previous code obfuscation on all these `return`?

Reply:
comments addressed